### PR TITLE
League post detail component

### LINKED ISSUE #70: Add League Post Detail Page Component
User Stories:

- #5

Components:

- #77 
- #78

### LINKED ISSUE #124: Add League Post Detail Service

### LINKED ISSUE #125: Add get league post detail by league post id

### PR DESCRIPTION
**Pull request by dudaskobendeguz**
- closes #70 
- closes #124 
- closes #125

### CODE REVIEW

File: angular/src/app/app-routing.module.ts [hunk 1]
```diff
@@ -4,14 +4,17 @@ import {UserPostContainerComponent} from "./components/user-post-components/user
...
   LeaguePostContainerComponent
 } from "./components/league-posts/league-post-container/league-post-container.component";
+import {LeaguePotsDetailComponent} from "./components/league-posts/league-pots-detail/league-pots-detail.component";
 import {UserPostDetailComponent} from "./components/user-post-components/user-post-detail/user-post-detail.component";
 import {ProfilePageComponent} from "./components/user/profile-page/profile-page.component";
 
 const routes: Routes = [
   {path: 'user-posts', component: UserPostContainerComponent},
   {path: 'league-posts', component: LeaguePostContainerComponent},
+  {path: 'league/:id', component: LeaguePotsDetailComponent},
```
Comment (benedekhalaj) [inline angular/src/app/app-routing.module.ts]: I think the 'all' and 'unique' paths should be the same

e.g. /user-posts, /user-posts/:id

File: angular/src/app/components/league-posts/league-pots-detail/league-pots-detail.component.css [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  background-color: #9ecce8;
+}
+
+mat-card.admin-player {
+  background-color: #9fb0b4;
+}
+
+mat-card.player:hover {
+  -moz-box-shadow:    3px 3px 5px 6px #ccc;
+  -webkit-box-shadow: 3px 3px 5px 6px #ccc;
```
Comment (benedekhalaj) [inline angular/src/app/components/league-posts/league-pots-detail/league-pots-detail.component.css:24]: epic css

File: angular/src/app/interfaces/league-details.ts [hunk 3]
```diff
@@ -7,7 +7,7 @@ export interface LeagueDetails {
   "image_source": string,
   "league_admin": User,
   "league_players": User[],
-  "number_of_finished_rounds": number,
-  "number_of_in_progress_rounds": number,
-  "number_of_rounds": number,
+  "numberOfFinishedRounds": number,
+  "numberOfInProgressRounds": number,
+  "numberOfRounds": number,
```
Comment (benedekhalaj) [inline angular/src/app/interfaces/league-details.ts:12]: if these fields are camelCase, what about image_source, league_admin, and so on?

File: angular/src/app/services/league-post-detail/league-post-detail.service.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+export class LeaguePostDetailService {
+  private usersUrl: string = '/api/leaguePostDetails';
+
+  constructor(
+    private messageService: MessageLoggerService,
+    private errorHandler: ErrorHandlerService,
+    private http: HttpClient
+  ) { }
+
+  getByLeagueId(id: number): Observable<LeagueDetails> {
```
Comment (benedekhalaj) [inline angular/src/app/services/league-post-detail/league-post-detail.service.ts:21]: Yeeeee

**Review by benedekhalaj — CHANGES_REQUESTED** [review]
Please look at my comments, might be useful :)

File: angular/src/app/app-routing.module.ts (hunk 1 above)
Comment (dudaskobendeguz) [reply angular/src/app/app-routing.module.ts]: ohh... okay

File: angular/src/app/interfaces/league-details.ts (hunk 3 above)
Comment (dudaskobendeguz) [reply angular/src/app/interfaces/league-details.ts:12]: fixed

File: angular/src/app/interfaces/league-details.ts [hunk 5]
```diff
@@ -2,11 +2,11 @@ import {User} from "./user";
...
   "name": string,
-  "image_source": string,
-  "league_admin": User,
-  "league_players": User[],
+  "imageSource": string,
+  "leagueAdmin": User,
+  "leaguePlayers": User[],
   "numberOfFinishedRounds": number,
   "numberOfInProgressRounds": number,
   "numberOfRounds": number,
```
Comment (benedekhalaj) [inline angular/src/app/interfaces/league-details.ts:12]: Ahh, finally

**Review by benedekhalaj — APPROVED** [review]

**Pull request merged**